cockatrice/src/window_main.cpp

```diff
@@ -296,6 +297,9 @@ void MainWindow::serverTimeout()
 void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32 endTime)
 {
     switch (r) {
+        case Response::RespClientUpdateRequired:
+            QMessageBox::critical(this, tr("Error"), tr("Your client does not contain features that the server requires, please update your client and try again."));
```

Comment:
I would translate the response container into an internal structure as early as possible. The method which generates the alert dialog could take a `QSet<QString>` of missing required features.